modules/apps/remote-app/remote-app-support-web/src/main/resources/META-INF/resources/index.js

```diff
@@ -0,0 +1,377 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const APPS_TO_IDS = new Map();
+
+const APP_IDS = new Set();
+
+const ERROR_CODES = {
+	/* eslint-disable sort-keys */
+	UNKNOWN: -100,
+	ALREADY_REGISTERED: -101,
+	/* eslint-enable sort-keys */
+};
+
+export const REMOTE_APP_PROTOCOL = 'com.liferay.remote.app.protocol';
+
+export const VERSION = 0;
+
+const RESPONSES = {};
+
+function escape(untrusted) {
```

Comment:
Can we use https://github.com/liferay/liferay-portal/blob/8518d2c9e0764f70f6ae6f25fec6d531827565ac/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/html_util.js ?

Reply:
Maybe! I'll let @wincent comment on that when he gets back from vacation :)

Reply:
We can, if this function even needs to survive. I only had this in here as a demo of dealing with "untrusted" input, but as noted in the PR description, the fact that we trust the remote app enough to proxy through any `fetch`/GraphQL request means that we effectively trust the remote pretty darn hard already.